src/views/AccountDrefApplications/ActiveDrefTable/index.tsx

```diff
@@ -0,0 +1,325 @@
+import { useCallback, useMemo, useState } from 'react';
+import { _cs, listToMap } from '@togglecorp/fujs';
+
+import Container from '#components/Container';
+import {
+    createStringColumn,
+    createExpandColumn,
+    createEmptyColumn,
+    createExpansionIndicatorColumn,
+    createDateColumn,
+    createElementColumn,
+} from '#components/Table/ColumnShortcuts';
+import { type RowOptions } from '#components/Table/types';
+import Table from '#components/Table';
+import Pager from '#components/Pager';
+import TableBodyContent from '#components/Table/TableBodyContent';
+import useDebouncedValue from '#hooks/useDebouncedValue';
+import useTranslation from '#hooks/useTranslation';
+import { numericIdSelector } from '#utils/selectors';
+import { useRequest } from '#utils/restRequest';
+import { hasSomeDefinedValue } from '#utils/common';
+
+import DrefTableActions, { type Props as DrefTableActionsProps } from '../DrefTableActions';
+import Filters, { type FilterValue } from '../Filters';
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+const NUM_ITEMS_PER_PAGE = 6;
+
+interface Props {
+    className?: string;
+    actions?: React.ReactNode;
+}
+
+function ActiveDrefTable(props: Props) {
+    const {
+        className,
+        actions,
+    } = props;
+
+    const strings = useTranslation(i18n);
+
+    const [page, setPage] = useState(1);
+    const [filterValue, setFilterValue] = useState<FilterValue>({
+        country: undefined,
+        type_of_dref: undefined,
+        disaster_type: undefined,
+        appeal_code: undefined,
+    });
+
+    const debouncedFilterValue = useDebouncedValue(filterValue);
+
+    const {
+        response: activeDrefResponse,
+        pending: activeDrefResponsePending,
+    } = useRequest({
+        url: '/api/v2/active-dref/',
+        query: {
+            offset: NUM_ITEMS_PER_PAGE * (page - 1),
+            limit: NUM_ITEMS_PER_PAGE,
+            ...debouncedFilterValue,
+        },
+        preserveResponse: true,
+    });
+
+    type DrefItem = NonNullable<NonNullable<typeof activeDrefResponse>['results']>[number];
+    type Key = DrefItem['id'];
+
+    const getLatestStageOfDref = useCallback(
+        (dref: DrefItem) => {
+            const {
+                final_report_details,
+                operational_update_details,
+                has_ops_update,
+                has_final_report,
+            } = dref;
+
+            if (has_final_report) {
+                type FinalReportType = typeof final_report_details;
+                const finalReportList = final_report_details as unknown as FinalReportType[];
```

Comment:
Why are we casting this array
Let's add NOTE

Reply:
We have notes slightly above

Reply:
Let's add NOTE anywhere we are casting data. It's easier and we don't miss it.